HW1/P6/P6.py

```diff
@@ -0,0 +1,123 @@
+import pyspark
+from pyspark import SparkContext, SparkConf
+sc = SparkContext()
+
+
+
+#read the file by lines
+data = sc.textFile('pg100.txt')
+
+#read the file by words
+data_split = data.flatMap(lambda s: s.split())
+
+def word_can_be_accepted(word):
+    '''get rid of a word in three situations:
+        contain only numbers,
+        contain only letters which are capitalized,
+        or contain only letters which are capitalized and end with a period.'''
+    if word.isdigit():
+        return False
+    if word.isupper():
+        return False
+    if word[-1]=='.' and word[:-1].isupper():
+        return False
+    return True
+data_check = data_split.filter(word_can_be_accepted)
```

Comment:
This will reject words, for example, that contain all capital letters and end with an exclamation point, which should be accepted.

```
In [1]: "TEST!".isupper()
Out[1]: True
```